packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -11,54 +11,122 @@ import { FoamFeature } from '../types';
+For example, check out some tabstops in the below list:
+
+- $1
+- $2
+- $3
+
+To create a note from this template, run the 'Foam: Create new note from template' command.
+`;
 
 async function getTemplates(): Promise<string[]> {
   const templates = await workspace.findFiles('.foam/templates/**.md');
-  // parse title, not whole file!
-  return templates.map(template => path.basename(template.path));
+  return templates.map(template => path.basename(template.fsPath));
+}
+
+async function offerToCreateTemplate(): Promise<void> {
+  const response = await window.showQuickPick(['Yes', 'No'], {
+    placeHolder:
+      'No templates available. Would you like to create one instead?',
+  });
+  if (response === 'Yes') {
+    commands.executeCommand('foam-vscode.create-new-template');
+    return;
+  }
+}
+
+async function createNoteFromTemplate(): Promise<void> {
+  const templates = await getTemplates();
+  if (templates.length === 0) {
+    return offerToCreateTemplate();
+  }
+  const activeFile = window.activeTextEditor?.document?.fileName;
+  const currentDir =
+    activeFile !== undefined
+      ? path.dirname(activeFile)
+      : workspace.workspaceFolders[0].uri.fsPath;
+  const selectedTemplate = await window.showQuickPick(templates);
+  if (selectedTemplate === undefined) {
+    return;
+  }
+
+  const defaultFileName = 'new-note.md';
+  const defaultDir = `${currentDir}${path.sep}${defaultFileName}`;
+  const filename = await window.showInputBox({
+    prompt: `Enter the filename for the new note`,
+    value: defaultDir,
+    valueSelection: [
+      defaultDir.length - defaultFileName.length,
+      defaultDir.length - 3,
+    ],
+    validateInput: value =>
+      value.length ? undefined : 'Please enter a value!',
+  });
+  if (filename === undefined) {
+    return;
+  }
+
+  const templateText = await workspace.fs.readFile(
+    Uri.file(`${templatesDir}/${selectedTemplate}`)
+  );
+  const snippet = new SnippetString(templateText.toString());
+  await workspace.fs.writeFile(
+    Uri.file(filename),
+    new TextEncoder().encode('')
+  );
+  await focusNote(filename, true);
+  await window.activeTextEditor.insertSnippet(snippet);
+}
+
+async function createNewTemplate(): Promise<void> {
+  const isFirstTemplate = (await getTemplates()).length === 0;
+  const defaultFileName = 'new-template.md';
+  const templatesDir = path.join(
+    workspace.workspaceFolders[0].uri.fsPath,
+    '.foam',
+    'templates',
+    defaultFileName
+  );
+  const filename = await window.showInputBox({
+    prompt: `Enter the filename for the new template`,
+    value: templatesDir,
+    valueSelection: [
+      templatesDir.length - defaultFileName.length,
+      templatesDir.length - 3,
+    ],
+    validateInput: value =>
+      value.length ? undefined : 'Please enter a value!',
+  });
+  if (filename === undefined) {
+    return;
+  }
+
+  const template = isFirstTemplate ? firstTemplateContent : '';
```

Comment:
Have you considered having a more generic template and using it all the times instead of just on first creation? Worst case they would just delete it, and best case it refreshes people's memory when they are creating new templates

Reply:
Good idea!